ndcli/dimcli/__init__.py

```diff
@@ -1607,12 +1607,20 @@ def print_tree(tree, level):
                 if 'children' in node:
                     print_tree(node['children'], level + 1)
 
-        layer3domains = [args.layer3domain]
-        if args.layer3domain == "all":
-            layer3domains = [l['name'] for l in self.client.layer3domain_list()]
+        if args.layer3domain == "all" or get_layer3domain(args.layer3domain) is None:
+            force_print_layer3domain = True
+            if args.container:
+                filters = dict(status=['Container'])
+                blocks = self.client.ipblock_get_attrs_multi(args.container, filters=filters)
```

Comment:
These two lines can result in objects being missed. The behavior before layer3domain=all was introduced the surrounding containers, subnets and the like were also returned. By searching only for containers, these might get missed and the output wrong.
But that should be found when changing the tests in a way that this situation is provoked.

Reply:
Not used anymore, iterating over the layer3domains instead.